xarray/core/coordinates.py

```diff
@@ -152,8 +153,32 @@ def to_index(self, ordered_dims: Sequence[Hashable] = None) -> pd.Index:
 
         return pd.MultiIndex(level_list, code_list, names=names)
 
+    def _drop_multiindexes(self, keys):
+        from .dataset import Dataset
+
+        for key in keys:
+            maybe_midx = self._data._indexes.get(key, None)
+            idx_coord_names = set(maybe_midx.index.names + [maybe_midx.dim])
+            if isinstance(maybe_midx, PandasMultiIndex):
+                warnings.warn(
+                    f"Updating MultiIndexed coordinate {key!r} would corrupt indices for "
+                    f"other variables: {list(maybe_midx.index.names)!r}. "
+                    f"This will raise an error in the future. Use `.drop_vars({idx_coord_names!r})` before "
+                    "assigning new coordinate values.",
+                    DeprecationWarning,
+                    stacklevel=3,
+                )
+                for k in idx_coord_names:
+                    if isinstance(self._data, Dataset):
+                        del self._data._variables[k]
+                        del self._data._indexes[k]
+                    else:
+                        del self._data._coords[k]
+                        del self._data._indexes[k]
+
```

Comment:
This seems ugly. Happy to take suggestions on how to do this in a cleaner fashion.